### PR TITLE
[CLOUDSTACK-9328]: Fix vlan issues from test suite test_privategw_acl.py in BVT

### PR DESCRIPTION
**Pull request by sanju1010**
Please refer to CLOUDSTACK-9328 for the details.

Test Results:
==========
test_01_vpc_privategw_acl (integration.smoke.test_privategw_acl.TestPrivateGwACL) ... === TestName: test_01_vpc_privategw_acl | Status : SUCCESS ===
ok

----------------------------------------------------------------------
Ran 1 test in 108.273s

OK

### CODE REVIEW

File: test/integration/smoke/test_privategw_acl.py [hunk 1]
```diff
@@ -287,6 +298,16 @@ def performVPCTests(self, vpc_off, restart_with_cleanup = False):
...
+            "select vnet from op_dc_vnet_alloc where physical_network_id=\
+            (select id from physical_network where uuid='%s' ) and taken is NULL;" % physical_networks[0].id
+        )
+        self.assertEqual(validateList(qresultset)[0],
+                         PASS,
+                         "Invalid sql query response"
+        )
+        #vlans = physical_networks[0].vlan.split('-')
+        vlans = qresultset
         vlans = physical_networks[0].vlan.split('-')
```
Comment (shwetaag) [inline test/integration/smoke/test_privategw_acl.py]: this split is not needed for getting correct vlan entry

**Comment by sanju1010** [comment]
Made changes as per Shwetaag's comments.

**Comment by alexandrelimassantana** [comment]
Does the _cleanup_resources(cls.api_client, cls._cleanup)_ cleans the db connection as well? I see that it is only 1 test as of now, but if there would be another, the setUp should also work correctly?

**Comment by shwetaag** [comment]
Did Code walk through . LGTM

**Comment by eriweb** [comment]
@remibergsma @wilderrodrigues @borisroman you guys have been testing a lot previously, would this change work for you?

**Comment by sanju1010** [comment]
@alexandrelimassantana , cleanup does not clean the db connection. So it should work for other tests as well.

**Comment by swill** [comment]
### CI RESULTS




**MULTIPLE FAILURES, NEEDS WORK!**

There are problems with this code.  Here is a summary of the issues.  I have included detailed results so you can track down the issues.

```
======================================================================
ERROR: test_02_vpc_privategw_static_routes (integration.smoke.test_privategw_acl.TestPrivateGwACL)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/data/git/cs1/cloudstack/test/integration/smoke/test_privategw_acl.py", line 263, in test_02_vpc_privategw_static_routes
    self.performVPCTests(vpc_off)
  File "/data/git/cs1/cloudstack/test/integration/smoke/test_privategw_acl.py", line 309, in performVPCTests
    vlan_1 = int(vlans[0])
TypeError: int() argument must be a string or a number, not 'tuple'
```

```
======================================================================
ERROR: test_03_vpc_privategw_restart_vpc_cleanup (integration.smoke.test_privategw_acl.TestPrivateGwACL)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/data/git/cs1/cloudstack/test/integration/smoke/test_privategw_acl.py", line 275, in test_03_vpc_privategw_restart_vpc_cleanup
    self.performVPCTests(vpc_off, True)
  File "/data/git/cs1/cloudstack/test/integration/smoke/test_privategw_acl.py", line 309, in performVPCTests
    vlan_1 = int(vlans[0])
TypeError: int() argument must be a string or a number, not 'tuple'
```

```
======================================================================
ERROR: test_04_rvpc_privategw_static_routes (integration.smoke.test_privategw_acl.TestPrivateGwACL)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/data/git/cs1/cloudstack/test/integration/smoke/test_privategw_acl.py", line 287, in test_04_rvpc_privategw_static_routes
    self.performVPCTests(vpc_off)
  File "/data/git/cs1/cloudstack/test/integration/smoke/test_privategw_acl.py", line 309, in performVPCTests
    vlan_1 = int(vlans[0])
TypeError: int() argument must be a string or a number, not 'tuple'
```

Not sure if this is related to one of the other failures or not...
```
======================================================================
FAIL: Test Remote Access VPN in VPC
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/data/git/cs1/cloudstack/test/integration/smoke/test_vpc_vpn.py", line 419, in test_01_vpc_remote_access_vpn
    vpnUser, "Failed to create Remote Access VPN User")
AssertionError: Failed to create Remote Access VPN User
```


**Associated Uploads**


* [results.txt](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1455/results.txt)


Uploads will be available until `2016-06-04 00:00:00 +0000 GMT`

*Comment created by [`upr comment`](https://github.com/swill/upr).*

**Comment by sanju1010** [comment]
@swill , I have made changes based on the CI results.

Following failure in the CI run is not related to the changes in this PR:
======================================================================
FAIL: Test Remote Access VPN in VPC
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/data/git/cs1/cloudstack/test/integration/smoke/test_vpc_vpn.py", line 419, in test_01_vpc_remote_access_vpn
    vpnUser, "Failed to create Remote Access VPN User")
AssertionError: Failed to create Remote Access VPN Us

**Comment by swill** [comment]
### CI RESULTS

```
Tests Run: 4
  Skipped: 0
   Failed: 0
   Errors: 0
```



**Associated Uploads**

**`/tmp/MarvinLogs/DeployDataCenter__Apr_28_2016_00_13_59_IGZ5H5:`**
* [dc_entries.obj](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1455/tmp/MarvinLogs/DeployDataCenter__Apr_28_2016_00_13_59_IGZ5H5/dc_entries.obj)
* [failed_plus_exceptions.txt](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1455/tmp/MarvinLogs/DeployDataCenter__Apr_28_2016_00_13_59_IGZ5H5/failed_plus_exceptions.txt)
* [runinfo.txt](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1455/tmp/MarvinLogs/DeployDataCenter__Apr_28_2016_00_13_59_IGZ5H5/runinfo.txt)

**`/tmp/MarvinLogs/test_privategw_acl_IEZB6Z:`**
* [failed_plus_exceptions.txt](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1455/tmp/MarvinLogs/test_privategw_acl_IEZB6Z/failed_plus_exceptions.txt)
* [results.txt](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1455/tmp/MarvinLogs/test_privategw_acl_IEZB6Z/results.txt)
* [runinfo.txt](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1455/tmp/MarvinLogs/test_privategw_acl_IEZB6Z/runinfo.txt)


Uploads will be available until `2016-06-27 02:00:00 +0200 CEST`

*Comment created by [`upr comment`](https://github.com/cloudops/upr).*

**Comment by swill** [comment]
I have tested this and I think we are ready.  Can you please do a force push to kick off jenkins again so we can get this PR all green?  Thanks...

**Comment by sanju1010** [comment]
@swill , it is all green now. Can you please merge this PR ?

**Comment by swill** [comment]
Yes, I think this one is ready...  Can I get one more LGTM code review on this one?  

@pdube @kiwiflyer @DaanHoogland @rafaelweingartner

**Comment by DaanHoogland** [comment]
@sanju1010 though the code looks alright I don't like the integration test should be snooping under the hood. I would rather see them behave as black-box tests. I am sure this kind of validating happens elsewhere but still I would like some reason for it, can you explain?

**Comment by swill** [comment]
@sanju1010 can you address @DaanHoogland's concerns?

**Comment by swill** [comment]
Bump...  @sanju1010 can you please follow up on @DaanHoogland's comments.  This one is ready to go otherwise, so it would be great if you could follow up with us on this @sanju1010.  Thx...

**Comment by swill** [comment]
@sanju1010 we have merge conflicts now.  Can you please rebase against the current master.  Thanks...

**Comment by swill** [comment]
@sanju1010 we have merge conflicts now. Can you please rebase against the current master. Code freeze is basically now, so if you want this in we need to get busy...

**Comment by sanju1010** [comment]
@swill, I have resolved the merge conflicts. Can you please push this to master?

**Comment by swill** [comment]
Master is currently frozen.  I am merging bug fixes if they are needed to stabilize master, but I don't think this qualifies...

**Comment by sanju1010** [comment]
@swill , can you consider marvin test related changes for merging into ACS master since they would not destabilize the master?

**Comment by sanju1010** [comment]
This will help in stabilizing CI runs.

**Comment by swill** [comment]
This is a bit of a grey area.  I will consider it, but I need jenkins and travis coming back clean to be comfortable with merging and I will have to run another CI on it since you fixed the merge conflicts.  You may have to close and reopen this PR a few times to get travis and jenkins to go green...

**Comment by swill** [comment]
@sanju1010 can you close and reopen this to kick off travis again?  Thx...

**Comment by bvbharatk** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 166
 Hypervisor xenserver
 NetworkType Advanced
 Passed=70
 Failed=3
 Skipped=3

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/yj3wnzbceo9uef2/AAB6u-Iap-xztdm6jHX9SjPja?dl=0


**Failed tests:**
* test_vpc_vpn.py

 * ContextSuite context=TestRVPCSite2SiteVpn>:setup Failing since 17 runs

 * ContextSuite context=TestVpcRemoteAccessVpn>:setup Failing since 17 runs

 * ContextSuite context=TestVpcSite2SiteVpn>:setup Failing since 17 runs


**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_static_role_account_acls
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_routers.py
test_reset_vm_on_reboot.py
test_snapshots.py
test_deploy_vms_with_varied_deploymentplanners.py
test_login.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_disk_offerings.py

**Comment by sanju1010** [comment]
Failures from CI are not related to changes in this file. This commit makes changes to test_privategw_acl.py

**Comment by bvbharatk** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 177
 Hypervisor xenserver
 NetworkType Advanced
 Passed=65
 Failed=1
 Skipped=3

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/yj3wnzbceo9uef2/AAB6u-Iap-xztdm6jHX9SjPja?dl=0


**Failed tests:**
* test_vm_life_cycle.py

 * ContextSuite context=TestVMLifeCycle>:setup Failing since 2 runs


**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_static_role_account_acls
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_vpc_vpn.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_routers.py
test_reset_vm_on_reboot.py
test_snapshots.py
test_deploy_vms_with_varied_deploymentplanners.py
test_login.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_disk_offerings.py

**Comment by swill** [comment]
### CI RESULTS

```
Tests Run: 85
  Skipped: 0
   Failed: 0
   Errors: 0
 Duration: 3h 55m 23s
```



**Associated Uploads**

**`/tmp/MarvinLogs/DeployDataCenter__Jul_18_2016_07_15_33_3UESMT:`**
* [dc_entries.obj](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1455/tmp/MarvinLogs/DeployDataCenter__Jul_18_2016_07_15_33_3UESMT/dc_entries.obj)
* [failed_plus_exceptions.txt](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1455/tmp/MarvinLogs/DeployDataCenter__Jul_18_2016_07_15_33_3UESMT/failed_plus_exceptions.txt)
* [runinfo.txt](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1455/tmp/MarvinLogs/DeployDataCenter__Jul_18_2016_07_15_33_3UESMT/runinfo.txt)

**`/tmp/MarvinLogs/test_network_33DRFL:`**
* [failed_plus_exceptions.txt](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1455/tmp/MarvinLogs/test_network_33DRFL/failed_plus_exceptions.txt)
* [results.txt](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1455/tmp/MarvinLogs/test_network_33DRFL/results.txt)
* [runinfo.txt](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1455/tmp/MarvinLogs/test_network_33DRFL/runinfo.txt)

**`/tmp/MarvinLogs/test_vpc_routers_5DK6QE:`**
* [failed_plus_exceptions.txt](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1455/tmp/MarvinLogs/test_vpc_routers_5DK6QE/failed_plus_exceptions.txt)
* [results.txt](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1455/tmp/MarvinLogs/test_vpc_routers_5DK6QE/results.txt)
* [runinfo.txt](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1455/tmp/MarvinLogs/test_vpc_routers_5DK6QE/runinfo.txt)


Uploads will be available until `2016-09-17 02:00:00 +0200 CEST`

*Comment created by [`upr comment`](https://github.com/cloudops/upr).*

**Comment by swill** [comment]
@sanju1010 can you close and reopen this PR to kick off travis again?

Thanks...

**Pull request merged**